vm/devices/storage/disk_nvme/nvme_driver/src/registers.rs

```diff
@@ -116,6 +126,9 @@ impl<T: DeviceRegisterIo + Inspect> Bar0<T> {
             if u32::from(csts) == !0 {
                 break false;
             }
+            if start.elapsed() >= timeout {
+                break false;
```

Comment:
The caller seems to rely on this actually resetting the device before it frees buffers. With this change, you might now return with the device still referencing these buffers. And that can lead to memory corruption.

Reply:
```
            // Hold onto responses until the reset completes so that waiting IOs do
            // not think the memory is unaliased by the device.
```